homeassistant/components/camera/synology.py

```diff
@@ -115,10 +124,10 @@ def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
     websession = aiohttp.ClientSession(
         loop=hass.loop, connector=connector, cookies={'id': session_id})
 
-    @asyncio.coroutine
+    @callback
     def _async_close_websession(event):
-        """Close webssesion on shutdown."""
-        yield from websession.close()
+        """Close websession on shutdown."""
+        websession.detach()

```

Comment:
If we use the connector from hass.websession, it will close by hass.websession.close. If we have a own connector for ssl, we close the connector in a own event and only detach the session here.